src/cfml/system/endpoints/Gist.cfc

```diff
@@ -0,0 +1,156 @@
+/**
```

Comment:
Overall, this file seems very very similar to the Git.cfc endpoint.  Let's try to simply extend the Git endpoint and override the differences.  See the `Github.cfc` and `Git_SSH.cfc` endpoints as examples that all extend the core Git endpoint and just make slight modifications to its behavior.  
I did a compare of the code in this CFC and the base Git endpoint and here are the big differences I found:
* Different name prefix set-- this is no issue. You'd still do this.
* Some URL manipulation on line 45-- no problem just do that in your base `resovlePackage()` method and then call the super to take over
* looks like you removed the commitish support which actually looks like a problem-- how would a specific commit even work now like your doc addition to the install command show?  I would suspect that should be able to work the same for Gist.
* Slightly different working in exception messages.  just change the base Git class to use `getNamePrefixes()` to make it dynamic.
* Custom `getDefaultName()`-- If this really needs to be diff then you can just override it in your Gist CFC.  Looking at the logic, it seems like it would still work OK for Gist since there wouldn't be slashes or .git in those package names.
* You removed the recent `getUpdate()`.  Not actually sure if you did that on purpose or if you just didn't have the latest dev branch, but I would expect the same tag names to apply here for someone who wanted to tag a Gist. npms does seem to show they support a semver tag
* You removed `getProtocol()` but you could just override this for the Gist endpoint and you could probably actually just use this method to append the entire `https://gist.github.com/` bit to the Gist ID!

Reply:
Thanks for the feedback. It does seem like there is a lot of code that can be reused. I will work on using the Git endpoint.

> looks like you removed the commitish support which actually looks like a problem

On line 48 I am checking to see if a commit-ish is passed. So passing a commit does work.

I was a little confused by this because you cannot really create branches or tags from the Gist UI which is why I took it out. I guess someone could branch and tag a gist using the Git client so it probably makes sense to support that.

I tested out passing a specific commit to NPM but it did not seem to be working. There docs say it should support commit-ish and semver but I am not sure that it actually does. Maybe a bug in NPM.

Reply:
Let me know when you have something to review again.  Since the logic in the base Git CFC supports both branches and commits, it should work fine in any case, and in the event that someone manages to get a branch in their Gist repo, then no problem.